imap_data_access/file_validation.py

```diff
@@ -444,11 +542,122 @@ def construct_path(self) -> Path:
             Upload path
         """
         spice_dir = imap_data_access.config["DATA_DIR"] / "spice"
-        subdir = _SPICE_DIR_MAPPING[self.file_extension]
+        subdir = _SPICE_DIR_MAPPING[self.spice_metadata["type"]]
         # Use the file suffix to determine the directory structure
         # IMAP_DATA_DIR/spice/<subdir>/filename
         return spice_dir / subdir / self.filename
 
+    @staticmethod
+    def _matches_on_group(
+        regex_list: Iterable[re.Pattern], string_to_parse: str
+    ) -> re.Match:
+        """Determine the first regular expression that matches the provided string.
+
+        Parameters
+        ----------
+        regex_list: list
+            A list of compiled regular expressions to be applied in order
+        string_to_parse: str
+                The string to check against the provided regular expressions
+
+        Returns
+        -------
+            The first match that satisfies a regular expression found in regex_list
+        """
+        for regex in regex_list:
+            m = regex.match(string_to_parse)
+            if m is not None:
+                return m
+        return None
+
+    @staticmethod
+    def _extract_parts(
+        filename: Path | str,
+        parts: list[str],
+        transforms: dict | None = None,
+        handle_missing_parts: bool = False,
+    ) -> dict | None:
+        """Extract the parts of a file.
+
+        Parameters
+        ----------
+        filename: Path | str
+            A filename to extract parts from
+        parts: list[str]
+            A list of parts to be extracted from the filename
+        transforms: dict[str: func]
+            A dictionary of group->function (that takes 1 string) to
+            transform the string into some other type
```

Comment:
This could be inlined I think rather than passed here? It looks like you pass a lambda later on, but could this just be fixed internally instead?

Reply:
Yeah I could do that rather than pass it in. I was trying to be flexible but I don't necessarily know under what scenario you'd ever want to transform a name into something different

Reply:
I prefer to limit options if they are unused, so definitely suggest removing parts and transforms here and do it all.

I'd also suggest removing the _match_on_groups and inline that here since this is the only place it is used I think?